### DISCUSSION TITLE
Magic Link: Redirect to Deployed URL

### DISCUSSION THREAD

**Discussion by cdt-eth** [supabase/supabase#2397] [Questions] [answered]
My Disney+ clone app is deployed via Netlify to [https://disnyplus.plus](https://disnyplus.plus)

When I click the Magic Link it only takes me to localhost:3000 which is fine in dev but in Prod I want it to go to my deployed site. 

<img width="664" alt="Screen Shot 2021-07-16 at 8 19 08 AM" src="https://user-images.githubusercontent.com/26611339/125946263-3de311cb-a3bc-405f-aa1b-6daae4ef9445.png">

Should the `Site URL` be _https://disnyplus.plus_ or the `Additional Redirect URLs`? Any way to have it so I can test with `localhost` in dev and then _https://disnyplus.plus_ in prod?

**Comment by kangmingtay** [comment, marked as answer]
Hey @christiandavidturner, i believe this is a bug with the gotrue-js library. The correct behavior for this should be setting your Site URL as https://disnyplus.plus and any additional redirect urls you want to whitelist (even for any other environments) in the "Additional Redirect URLs". I've created an issue [here](https://github.com/supabase/gotrue-js/issues/104) so we can address it in the future. 

If you're using the supabase api library for javascript, it will not work for now. So an alternative would be to create an api endpoint using fetch or axios to hit the magiclink endpoint. I've provided a sample curl command for you to format your fetch / axios request: 

```
curl -X POST 'https://<PROJECT_REF>.supabase.co/auth/v1/magiclink?redirect_to="http://localhost:9999"' \
-H "apikey: SUPABASE_KEY" \
-H "Content-Type: application/json" \
-d '{
  "email": "someone@email.com"
}'
```

Let me know if this helps!

**Reply by kangmingtay** [reply]
Update: I've tried using axios to do this and it would look something like this:
```
const resp = await axios({
            method: 'post',
            headers: {
                'apikey': REACT_APP_SUPABASE_KEY,
                'Content-Type': 'application/json'
            },
            baseURL: REACT_APP_SUPABASE_URL,
            url: "/auth/v1/magiclink",
            params: {
                redirect_to: "YOUR_REDIRECT_URL"
            },
            data: {
                email: email
            }
        })
```

Make sure you whitelist the url you want to redirect to under "Additional Redirect URLs". They have to be an exact match too.

**Reply by xerod** [reply]
The javascript library is working now, something like this would work:

```
supabase.auth.signIn(data, { redirectTo: SUPABASE_REDIRECT_URL })
```
